homeassistant/components/teslemetry/number.py

```diff
@@ -0,0 +1,201 @@
+"""Number platform for Teslemetry integration."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from itertools import chain
+
+from tesla_fleet_api.const import Scope
+
+from homeassistant.components.number import (
+    NumberDeviceClass,
+    NumberEntity,
+    NumberEntityDescription,
+    NumberMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, PRECISION_WHOLE, UnitOfElectricCurrent
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.icon import icon_for_battery_level
+
+from .entity import TeslemetryEnergyInfoEntity, TeslemetryVehicleEntity
+from .models import TeslemetryEnergyData, TeslemetryVehicleData
+
+
+@dataclass(frozen=True, kw_only=True)
+class TeslemetryNumberEntityDescription(NumberEntityDescription):
+    """Describes Teslemetry Number entity."""
+
+    func: Callable
+    native_min_value: float
+    native_max_value: float
+    min_key: str = ""
+    max_key: str = ""
```

Comment:
Could be `None` instead to bail early instead of looking for a key in the data each time?

Reply:
Done that and refactored the descriptions a bit to avoid mypy issues.